pkg/controller/controller_binding.go

```diff
@@ -117,12 +134,20 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 				binding.Namespace,
 				binding.Name,
 			)
-			return nil
+			return true, nil
 		}
 	}
 
 	glog.V(4).Infof("Processing ServiceInstanceCredential %v/%v", binding.Namespace, binding.Name)
 
+	toUpdate, err := makeServiceInstanceCredentialClone(binding)
+	if err != nil {
+		return false, err
+	}
+
+	c.setServiceInstanceCredentialForPendingChange(toUpdate)
+	defer c.updateServiceInstanceCredentialStatus(toUpdate)
```

Comment:
why defer this call?

Reply:
The function will be doing a number of changes to the binding. The status for the binding will be updated in the catalog after all the changes are done. The alternative to doing a defer is to riddle the call to update the status in many, many places in the function and have to worry about remembering to add it in with every new return we add to the function.